### PR TITLE
Install SMT solvers for CI testing

### PR DESCRIPTION
**Pull request by wadoon**
Currently, no SMT-solver is available during the CI build and test. 

This MR brings a small bash scripts, which download and "install" z3 and CVC5. Both SMT-solvers should be available via `$PATH`. Caching is activated. 

The script uses the `gh`, the cli application for/from Github, to download the latest releases.

### CODE REVIEW

**Comment by wadoon** [comment]
/cc @mattulbrich @jwiesler 

@WolframPfeifer and I have pinpoint the current problems on the CI tests.

1. On other PRs two proofs in the  "redux" (`automaticJAVADL.txt`) are failing due to the missing of z3. 
1. This PR installs z3 and cvc5. The "redux" is tested successfully. But unit-tests are aborted maybe due to resource restriction. 

On Jenkins, we can observe that "KeY Master" takes 5 hours. 

Currently, this PR only installs SMT for integration tests.

**Comment by wadoon** [comment]
PR is ready. The remaining red cross is due to failing test reporting (because of permission management issues/branch from different repo).

**Review by mattulbrich — DISMISSED** [review]

File: .github/dlsmt.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/sh
+
```
[the remark this answers is not in the document]
Comment (wadoon) [reply .github/dlsmt.sh:2]: Comments added, but this revoked the approval.

**Comment by mi-ki** [comment]
Installing SMT solvers for testing is a great idea! Yet, always taking the most recent release might not be sensible, and the version(s) of the respective solver(s) should be consistent in order to allow reproducibility.

**Comment by mattulbrich** [comment]
Fixing a specific version may prevent us from running into unexpected failures. Right.
But: How do we find out if latest versions do not work with KeY any longer. Also quite important to learn.

**Comment by mi-ki** [comment]
> Fixing a specific version may prevent us from running into unexpected failures. Right. But: How do we find out if latest versions do not work with KeY any longer. Also quite important to learn.

I was not proposing to fix a specific version for eternity. Naturally, the version(s) should be updated when it becomes sensible. However, for a specific KeY version, the supported solver version(s) should be reliably tested.

**Review by mattulbrich — APPROVED** [review]

**Review by mattulbrich — APPROVED** [review]
Let's merge this since main is ready since main is red b/c of this.

But yes, let's fix a Z3 version then.

**Pull request merged**